frontend/components/Filter/FilterBanner.tsx

```diff
@@ -225,16 +206,24 @@ export default function FilterBanner({
     setEventDataList([...filteredEventDataList]);
     closeModal();
   }
+
   useEffect(() => {
-    applyFilters(selectedSport);
+    if (triggerFilterApply !== undefined) {
```

Comment:
add comment here to clarify that this will only run after the first render - because as what i understand that is what this is doing